app/src/main/java/org/thoughtcrime/securesms/groups/GroupMembersViewModel.kt

```diff
@@ -1,30 +1,58 @@
 package org.thoughtcrime.securesms.groups
 
 import android.content.Context
+import android.content.Intent
+import androidx.lifecycle.viewModelScope
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
 import dagger.hilt.android.lifecycle.HiltViewModel
 import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import org.session.libsession.database.StorageProtocol
+import org.session.libsession.utilities.Address
 import org.session.libsession.utilities.ConfigFactoryProtocol
 import org.session.libsession.utilities.UsernameUtils
 import org.session.libsignal.utilities.AccountId
+import org.thoughtcrime.securesms.conversation.v2.ConversationActivityV2
 import org.thoughtcrime.securesms.util.AvatarUtils
 
 
 @HiltViewModel(assistedFactory = GroupMembersViewModel.Factory::class)
 class GroupMembersViewModel @AssistedInject constructor(
     @Assisted private val groupId: AccountId,
-    @ApplicationContext context: Context,
-    storage: StorageProtocol,
+    @ApplicationContext private val context: Context,
+    private val storage: StorageProtocol,
     configFactory: ConfigFactoryProtocol,
     usernameUtils: UsernameUtils,
     avatarUtils: AvatarUtils
 ) : BaseGroupMembersViewModel(groupId, context, storage, usernameUtils, configFactory, avatarUtils) {
 
+    private val _navigationActions = Channel<Intent>()
+    val navigationActions = _navigationActions.receiveAsFlow()
```

Comment:
The flow will only be created once for every UI out there, is that the expected behavior? If the UI unsubscribes I think the flow is then terminated? Should we use this instead?

```suggestion
    val navigationActions get() = _navigationActions.receiveAsFlow()
```